### PR TITLE
Add precondition skipping to documentation

### PR DESCRIPTION
**Pull request by prikha**
Precondition block can be halted if you are fine with the first failure only

Describe the changes and motivations for the pull request.

### Review

- [x] Document code according to [Getting Started with Yard](http://www.rubydoc.info/gems/yard/file/docs/GettingStarted.md).
- [x] All tests are passing.
- [x] Test manually.
- [x] Get approval.

### Pre-merge checklist

- [x] The PR relates to a single subject with a clear title and description in grammatically correct, complete sentences.
- [x] Verify that feature branch is up-to-date with `master` (if not - rebase it).
- [x] Double check the quality of [commit messages](http://chris.beams.io/posts/git-commit/).
- [x] Squash related commits together.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -268,6 +268,16 @@ precondition if: -> { subject.active? } do
 end
 ```
 
+Failing one precondition doesn't stop the execution. So if you have multiple conditions defined all of the would be checked. To stop propagation use `next` like this:
```
Comment (piotrimbierowicz) [inline docs/index.md]: `all of them will be checked` ?

Comment (piotrimbierowicz) [reply docs/index.md]: `like in this example` ?

File: docs/index.md [hunk 2]
```diff
@@ -268,6 +268,16 @@ precondition if: -> { subject.active? } do
 end
 ```
 
+Failing one precondition doesn't stop the execution. So if you have multiple conditions defined all of the would be checked. To stop propagation use `next` like this: 
+
+```
+precondition do
+  next decline_with(:failing_first) unless first_condition_is_false?
+  break decline_with(:failing_second) unless first_condition_is_true_but_second_is_false?
```
Comment (piotrimbierowicz) [inline docs/index.md]: What does the `break` do here?

Comment (prikha) [reply docs/index.md]: It is an example. NP - the docs are rejected by the Granite guys

**Pull request closed without merging**